docs/batch_inference_with_mms.md

```diff
@@ -0,0 +1,335 @@
+        ...
+...
+    def inference(self, model_input):
+        Batch = namedtuple('Batch', ['data'])
+        if self.error is not None:
+            return None
+
+        self.mx_model.forward(Batch([model_input]), is_train=False)
+        outputs = self.mx_model.get_outputs()
+        for d in outputs:
+            if isinstance(d, list):
+                for n in outputs:
+                    if isinstance(n, mx.ndarray.ndarray.NDArray):
+                        n.wait_to_read()
+            elif isinstance(d, mx.ndarray.ndarray.NDArray):
+                d.wait_to_read()
+        res = mx.ndarray.split(outputs[0], axis=0, num_outputs=outputs[0].shape[0])
+        res = [res] if not isinstance(res, list) else res
+        return res
+...    
+``` 
+
+Lets move onto post-processing. 
+
+ 
+#### Postprocess logic
+The output of inference logic is fed to the post-processing logic. As we saw before, during preprocessing, if there aren't `batch_size` number of requests, preprocessing logic
+pads the difference with 0's. These are artificially created requests sent to the module for inference. **In post-process we should ignore these artificially padded requests**. 
+Lets look at the sample logic
+
+```python
+import mxnet as mx
+from collections import namedtuple
+
+class BatchingExample(object):
+...
+    def initialize(self, context):
+        ...
+...
+    def preprocess(self, request):
+        ...
+...
+    def inference(self, model_input):
+        ...
+    def postprocess(self, data):
+        if self.error is not None:
+            return [self.error] * self._batch_size
+
+        res = [self.top_probability(d, self.labels, top=5) for d in data[:self._num_requests]]
+        return res
+...
+```
+
+In the above code, we iterate only until **self._num_requests**. This variable is assigned a value during the preprocessing step. This logic ensures that the postprocess logic is run only
+for the actual requests coming from external clients.
+
+
+### MMS Configuration
+MMS gets the batch-size information from **POST /models** API. For more information about this API please refer [management API document](https://github.com/awslabs/mxnet-model-server/blob/master/docs/management_api.md).
+The configuration that we are interested in is the following: 
+1. `batch_size`: This is the maximum batch size that a model is expected to handle. 
+2. `max_batch_delay`: This is the maximum batch delay time MMS frontend waits to receive `batch_size` number of requests. If MMS doesn't receive `batch_size` number of requests
+before this timer time's out, it sends what ever requests that were received to the model `handler`.
+ 
+These configurations are used both in MMS Frontend and the handler code.  
+in the model's custom-service-code (a.k.a the handler code).
+1. MMS Frontend: MMS Frontend associates the batch related configuration with each model. The frontend then tries to aggregate the batch-size number of requests and send it to the backend.
```

Comment:
MMS Frontend is an internal terminology, better just use "MMS" to avoid confusion to user.